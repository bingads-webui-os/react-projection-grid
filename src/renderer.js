import React from 'react';
import PropTypes from 'prop-types';

function renderTrs(trs) {
  return (
    trs.map(tr => (
      <tr key={tr.key} {...tr.attributes}>
        {tr.tds.map(td => (
          <td key={td.key} {...td.attributes}>
            <td.content.Component {...td.content.props} {...td.content.events} />
          </td>
        ))}
      </tr>
    ))
  );
}

export const TableRender = (props) => {
  const { table } = props.model;
  return (
    <div>
      <table>
        <caption {...table.caption.attributes}>
          <table.caption.content.Component
            {...table.caption.content.props}
            {...table.caption.content.events}
          />
        </caption>
        {table.colgroups.map(colgroup => (
          <colgroup key={colgroup.key} {...colgroup.attributes}>
            {colgroup.cols.map(col => (
              <col key={col.key} {...col.attributes} />
            ))}
          </colgroup>
        ))}

        <thead {...table.thead.attributes}>
          {table.thead.trs.map(tr => (
            <tr key={tr.key} {...tr.attributes}>
              {tr.ths.map(th => (
                <th key={th.key} {...th.attributes}>
                  <th.content.Component {...th.content.props} {...th.content.events} />
                </th>
              ))}
            </tr>))}
        </thead>
        {table.tbodies.map(tbody => (
          <tbody key={tbody.key} {...tbody.attributes}>
            {renderTrs(tbody.trs)}
          </tbody>
        ))}
        <tfoot {...table.tfoot.attributes}>
          {renderTrs(table.tfoot.trs)}
        </tfoot>
      </table>
    </div>
  );
};

TableRender.defaultProps = {
  model: {},
};

TableRender.propTypes = {
  model: PropTypes.oneOfType([
    PropTypes.object,
  ]).isRequired,
};
