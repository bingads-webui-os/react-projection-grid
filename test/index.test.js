import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import ProjectionGridReact from '../src/index';

const mockConfig = {
  data: [{ UserName: 'abc' }],
  primaryKey: 'UserName',

};

configure({ adapter: new Adapter() });

describe('ProjectionGridReact', () => {
  test('should not throw any errors', () => {
    expect(() => {
      renderer.create(<ProjectionGridReact {...mockConfig} />);
    }).not.toThrow();
  });

  test('should render correctly', () => {
    const grid = renderer.create(<ProjectionGridReact {...mockConfig} />);

    const tree = grid.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
