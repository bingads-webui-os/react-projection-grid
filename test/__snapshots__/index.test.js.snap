// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectionGridReact should render correctly 1`] = `
<div>
  <table
    className=""
    style={Object {}}
  >
    <caption
      className=""
      style={Object {}}
    />
    <thead
      className=""
      style={Object {}}
    >
      <tr
        className=""
        style={Object {}}
      />
    </thead>
    <tbody
      className=""
      style={Object {}}
    >
      <tr
        className=""
        style={Object {}}
      />
    </tbody>
    <tfoot />
  </table>
</div>
`;
